bionemo-recipes/recipes/llama3/tests/test_dataset.py

```diff
@@ -0,0 +1,295 @@
+
+
+def test_streaming_dataset_produces_batches(tokenizer_path, simple_parquet):
+    """Test that streaming mode works and produces valid batches."""
+    distributed_config = DistributedConfig(rank=0, world_size=1)
+    
+    load_dataset_kwargs = {
+        "path": "parquet",
+        "data_files": simple_parquet,
+        "split": "train",
+        "streaming": True,
+    }
+    
+    dataloader, _ = create_bshd_dataloader(
+        distributed_config=distributed_config,
+        tokenizer_path=tokenizer_path,
+        load_dataset_kwargs=load_dataset_kwargs,
+        micro_batch_size=2,
+        num_workers=0,
+        max_seq_length=500,
+        stride=100,
+        buffer_size=10_000,
+    )
+    
+    # Get a batch
+    batch = next(iter(dataloader))
+    
+    # Check batch structure
+    assert "input_ids" in batch
+    assert "attention_mask" in batch
+    assert "labels" in batch
+    
+    # Check batch is tensors
+    assert isinstance(batch["input_ids"], torch.Tensor)
+    assert batch["input_ids"].shape[0] == 2
+
+
+def test_windowing_in_dataset_creates_multiple_samples(tokenizer_path, tmp_path):
+    """Test that the dataset's windowing creates expected number of samples."""
+    # Create a 3kbp sequence
+    parquet_path = tmp_path / "genomic_sequences.parquet"
+    sequence = "A" * 3000
+    table = pa.table({"sequence": [sequence]})
+    pq.write_table(table, parquet_path)
+    
+    distributed_config = DistributedConfig(rank=0, world_size=1)
+    
+    load_dataset_kwargs = {
+        "path": "parquet",
+        "data_files": str(parquet_path),
+        "split": "train",
+    }
+    
+    tokenized_dataset, tokenizer = create_tokenized_dataset(
+        distributed_config=distributed_config,
+        tokenizer_path=tokenizer_path,
+        load_dataset_kwargs=load_dataset_kwargs,
+        max_seq_length=1000,
+        stride=800,  # 800 token overlap, so 200 token step
+        buffer_size=10_000,
+        use_lazy_tokenization=False,  # Use eager tokenization to expand windows
+    )
+    
+    # Count samples
+    num_samples = len(tokenized_dataset)
+    
+    # With 3000bp sequence, max_length=1000, stride=800 (overlap)
+    # Should create multiple windows (at least 2)
+    assert num_samples >= 2, f"Expected at least 2 windows, got {num_samples}"
+
+
+def test_lazy_tokenization_returns_batch(tokenizer_path, simple_parquet):
+    """Test that lazy tokenization works and returns valid batches."""
+    distributed_config = DistributedConfig(rank=0, world_size=1)
+    
+    load_dataset_kwargs = {
+        "path": "parquet",
+        "data_files": simple_parquet,
+        "split": "train",
+        "streaming": False,
+    }
+    
+    dataloader, _ = create_bshd_dataloader(
+        distributed_config=distributed_config,
+        tokenizer_path=tokenizer_path,
+        load_dataset_kwargs=load_dataset_kwargs,
+        micro_batch_size=2,
+        num_workers=0,
+        max_seq_length=500,
+        stride=100,
```

Comment:
you could add a test that checks the stride stuff, right? maybe with tokens like

sequence = 1000 * "A" + 1000 * "B"
and then make sure that sequences come out the way we expect